pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -1060,7 +1060,29 @@ func isKernelPid(pid int) bool {
 	return err != nil && os.IsNotExist(err)
 }
 
+// GetCapacity returns node capacity data for "cpu", "memory", "ephemeral-storage", and "huge-pages*"
+// At present this method is only invoked when introspecting ephemeral storage
 func (cm *containerManagerImpl) GetCapacity() v1.ResourceList {
+	// We store allocatable ephemeral-storage in the capacity property once we Start() the container manager
+	if _, ok := cm.capacity[v1.ResourceEphemeralStorage]; !ok {
```

Comment:
@ehashman this modified implementation attempts a single "best effort" call to cAdvisor if ephemeral storage data is missing, which will cap any additional delays to node registration to the amount of time it takes to get a response from `cadvisorInterface.RootFsInfo()` (plus some additional accounting for ResourceList data construction in the successful data retrieval cases).

Also, rather than returning an empty ResourceList in the "failed to get data" case, we will simply return the capacity property, which mimics the current behavior.

Reply:
I see you had retry logic before and now remove it. So in your test, did you see any cases that cadvisor might not be initialized at this moment and return no data. If so, how much time typically we need to wait for it to be ready? Thanks!

Reply:
In my tests I *did not* encounter cAdvisor not being ready (though I didn't run like 10,000 real-world simulations). So, I think in practice not retrying in a loop will work in a very high percentage of cases. And the edge case scenarios (expressed in unit tests) are essentially no-ops for a kubelet runtime, and functionally equivalent to what happens today.

By the way, as you probably figured out, I removed the retries based on prior review feedback.

Reply:
just confirm retry is not needed here? I still see some value of retrying in no inconsistent data will be returned from cadvisor?